packages/client/src/open-feature.ts

```diff
@@ -87,9 +87,8 @@ export class OpenFeatureAPI extends OpenFeatureCommonAPI {
       this._provider = provider;
       this._providerReady = false;
 
-      if (!this._provider.events) {
-        this._provider.events = new OpenFeatureEventEmitter();
-      }
+      this.removeListeners(oldProvider);
+      this.attachListeners(this._provider);
```

Comment:
thanks to these now the event tests pass also without reinitializing the API, the only exception is ```
Requirement 5.1.1', () => {
    it('The provider defines a mechanism for signaling the occurrence of an event ''PROVIDER_READY`'
```
The ready is not handled if we register it after... 🦆 do you think it is worth investigating why?

Reply:
@RealAnna OK so I found a bug with my implementation. I was still emitting the events directly from the provider events object. That's a problem, because some providers wont implement events, but as the recent spec change dictates, READY/ERROR events on the client must STILL must fire in the case on the initialization. For more details on that, see [here](https://github.com/open-feature/spec/pull/182/files#diff-daf30d840f7e7b1752899ac2c4b268af214e6c8e7058549c8d4218abd2e86ba3R104-R129). I added tests for this case.

I think what you were pointing out was an issue because we were adding handlers directly to the provider events. I know we chose to do this initially because it scoped the tests more (unit style instead of integration style) but I think this isn't ideal. The spec doesn't actually say that it needs to be possible to subscribe to provider events - the provider just needs to be able to emit them. The important thing to test is that handlers run on the `client` when those events are emitted. I have updated the tests in question to work that way (some already did work that way).

See my changes [here](https://github.com/open-feature/js-sdk/pull/422/commits/1b447a853a4233455b7776eea2fe19cf1f121cf2).

Reply:
@toddbaert I see your point, this makes a lot more sense to me now!